vignettes/sim_fixed_design_custom.Rmd

```diff
@@ -0,0 +1,343 @@
+  ) 
+
+sim_res |> gt() |> tab_header("One Simulation Results")
+```
+
+
+# Step 4: Perform the above single simulation repeatedly
+
+We will now merge Steps 1 to 3 into a single function named `one_sim()`, which facilitates a single simulation run. The construction of `one_sim()` involves copying all the lines of code from Steps 1 to 3.
+
+```{r}
+one_sim <- function(sim_id = 1, 
+                    # arguments from Step 1: design characteristic
+                    n, stratum, enroll_rate, fail_rate, dropout_rate, block, 
+                    # arguments from Step 2； cutting method
+                    min_n_overall, min_followup,
+                    # arguments from Step 3； testing method
+                    fh, mb, xu, rmst, ms, mc
+                    ) {
+    # Step 1: simulate a time-to-event data
+    uncut_data <- sim_pw_surv(
+      n = n,
+      stratum = stratum,
+      block = block,
+      enroll_rate = enroll_rate,
+      fail_rate = fail_rate,
+      dropout_rate = dropout_rate) 
+    
+    ## Step 2: Cut data
+    cut_date <- get_analysis_date(min_n_overall = min_n_overall, min_followup = min_followup, data = uncut_data)
+    cut_data <- uncut_data |> cut_data_by_date(cut_date)
+    
+    # Step 3: Run tests
+    sim_res_lr <- cut_data |> wlr(weight = fh(rho = 0, gamma = 0))
+    sim_res_fh <- cut_data |> wlr(weight = fh(rho = fh$rho, gamma = fh$gamma))
+    sim_res_mb <- cut_data |> wlr(weight = mb(delay = mb$delay, w_max = mb$w_max))
+    sim_res_xu <- cut_data |> wlr(weight = early_zero(early_period = xu$early_period))
+    sim_res_rmst <- cut_data |> rmst(tau = rmst$tau)
+    sim_res_ms <- cut_data |> milestone(ms_time = ms$ms_time)
+    sim_res_mc <- cut_data |> maxcombo(rho = mc$rho, gamma = mc$gamma)
+    
+    sim_res <- tribble(
+      ~`Sim ID`, ~Method, ~Parameter, ~Z, ~Estimate, ~SE, ~`P value`,
+      sim_id, sim_res_lr$method, sim_res_lr$parameter, sim_res_lr$z, sim_res_lr$estimate, sim_res_lr$se, pnorm(-sim_res_lr$z),
+      sim_id, sim_res_fh$method, sim_res_fh$parameter, sim_res_fh$z, sim_res_fh$estimate, sim_res_fh$se, pnorm(-sim_res_fh$z),
+      sim_id, sim_res_mb$method, sim_res_mb$parameter, sim_res_mb$z, sim_res_mb$estimate, sim_res_mb$se, pnorm(-sim_res_mb$z),
+      sim_id, sim_res_xu$method, sim_res_xu$parameter, sim_res_xu$z, sim_res_xu$estimate, sim_res_xu$se, pnorm(-sim_res_xu$z),
+      sim_id, sim_res_rmst$method, sim_res_rmst$parameter|> as.character(), sim_res_rmst$z, sim_res_rmst$estimate, sim_res_rmst$se, pnorm(-sim_res_rmst$z),
+      sim_id, sim_res_ms$method, sim_res_ms$parameter |> as.character(), sim_res_ms$z, sim_res_ms$estimate, sim_res_ms$se, pnorm(-sim_res_ms$z),
+      sim_id, sim_res_mc$method, sim_res_mc$parameter, NA, NA, NA, sim_res_mc$p_value
+  ) 
+      
+    return(sim_res)
+}
+```
+
+After that, we will execute `one_sim()` multiple times using parallel computation. The following lines of code takes 4 works to run 100 simulations. 
+```{r}
+set.seed(2025)
```

Comment:
```suggestion
plan("multisession", workers = 2)
set.seed(2025)
```

It looks like you accidentally deleted the lines where you specific the number of parallel workers when you were removing the timing code.

Reply:
Thanks for catching the above. These have been fixed.